File: snippets/go/pay_onchain.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func GetCurrentLimits() {
	// ANCHOR: get-current-reverse-swap-limits
	if currentLimits, err := sdk.OnchainPaymentLimits(); err == nil {
		log.Printf("Minimum amount, in sats: %v", currentLimits.MinSat)
		log.Printf("Maximum amount, in sats: %v", currentLimits.MaxSat)
	}
	// ANCHOR_END: get-current-reverse-swap-limits
}

func PreparePayOnchain(currentLimits breez_sdk.OnchainPaymentLimitsResponse) {
	// ANCHOR: prepare-pay-onchain
	sendAmountSat := currentLimits.MinSat
	satPerVbyte := uint32(10)

	req := breez_sdk.PrepareOnchainPaymentRequest{
		AmountSat:      sendAmountSat,
		AmountType:     breez_sdk.SwapAmountTypeSend,
		ClaimTxFeerate: satPerVbyte,
	}

	if prepareRes, err := sdk.PrepareOnchainPayment(req); err == nil {
		log.Printf("Sender amount, in sats: %v", prepareRes.SenderAmountSat)
		log.Printf("Recipient amount, in sats: %v", prepareRes.RecipientAmountSat)
		log.Printf("Total fees, in sats: %v", prepareRes.TotalFees)
	}
	// ANCHOR_END: prepare-pay-onchain
}

func StartReverseSwap(prepareRes breez_sdk.PrepareOnchainPaymentResponse) {
	// ANCHOR: start-reverse-swap
	destinationAddress := "bc1.."

	payOnchainRequest := breez_sdk.PayOnchainRequest{
		RecipientAddress: destinationAddress,
		PrepareRes:       prepareRes,
	}

	if reverseSwapInfo, err := sdk.PayOnchain(payOnchainRequest); err == nil {
		log.Printf("%#v", reverseSwapInfo)
	}
	// ANCHOR_END: start-reverse-swap
}

func CheckReverseSwapStatus() {
	// ANCHOR: check-reverse-swaps-status
	if swaps, err := sdk.InProgressOnchainPayments(); err == nil {
		for _, swap := range swaps {
			log.Printf("Onchain payment %v in progress, status is %v", swap.Id, swap.Status)
		}
	}
	// ANCHOR_END: check-reverse-swaps-status
}

File: snippets/dart_snippets/lib/pay_onchain.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<OnchainPaymentLimitsResponse> getCurrentLimits() async {
  // ANCHOR: get-current-reverse-swap-limits
  OnchainPaymentLimitsResponse currentLimits = await breezSDK.onchainPaymentLimits();
  print("Minimum amount, in sats: ${currentLimits.minSat}");
  print("Maximum amount, in sats: ${currentLimits.maxSat}");
  print("Maximum payable, in sats: ${currentLimits.maxPayableSat}");
  // ANCHOR_END: get-current-reverse-swap-limits
  return currentLimits;
}

Future<PrepareOnchainPaymentResponse> preparePayOnchain({
  required int amountSat,
  required int satPerVbyte,
}) async {
  // ANCHOR: prepare-pay-onchain
  PrepareOnchainPaymentRequest req = PrepareOnchainPaymentRequest(
    amountSat: amountSat,
    amountType: SwapAmountType.Send,
    claimTxFeerate: satPerVbyte,
  );
  PrepareOnchainPaymentResponse prepareRes = await breezSDK.prepareOnchainPayment(req: req);
  print("Sender amount: ${prepareRes.senderAmountSat} sats");
  print("Recipient amount: ${prepareRes.recipientAmountSat} sats");
  print("Total fees: ${prepareRes.totalFees} sats");
  // ANCHOR_END: prepare-pay-onchain
  return prepareRes;
}

Future<PayOnchainResponse> startReverseSwap({
  required String onchainRecipientAddress,
  required PrepareOnchainPaymentResponse prepareRes,
}) async {
  // ANCHOR: start-reverse-swap
  PayOnchainRequest req = PayOnchainRequest(
    recipientAddress: onchainRecipientAddress,
    prepareRes: prepareRes,
  );
  PayOnchainResponse res = await breezSDK.payOnchain(req: req);
  // ANCHOR_END: start-reverse-swap
  return res;
}

Future<List<ReverseSwapInfo>> checkReverseSwapStatus() async {
  // ANCHOR: check-reverse-swaps-status
  List<ReverseSwapInfo> inProgOnchainPaymentList = await breezSDK.inProgressOnchainPayments();
  for (var inProgOnchainPayment in inProgOnchainPaymentList) {
    print("Onchain payment ${inProgOnchainPayment.id} in progress, status is ${inProgOnchainPayment.status.name}");
  }
  // ANCHOR_END: check-reverse-swaps-status
  return inProgOnchainPaymentList;
}
